pkg/services/gitops/install.go

```diff
@@ -229,3 +259,47 @@ func (g *Gitops) applyManifestsToK8s(ctx context.Context, namespace string, mani
 
 	return nil
 }
+
+const LabelPartOf = "app.kubernetes.io/part-of"
+
+var ErrNamespaceNotFound = errors.New("namespace not found")
+
+func (g *Gitops) fetchNamespaceWithLabel(ctx context.Context, key string, value string) (string, error) {
+	selector := labels.NewSelector()
+
+	partOf, err := labels.NewRequirement(key, selection.Equals, []string{value})
+	if err != nil {
+		return "", fmt.Errorf("bad requirement: %w", err)
+	}
+
+	selector = selector.Add(*partOf)
+
+	options := client.ListOptions{
+		LabelSelector: selector,
+	}
+
+	nsl := &corev1.NamespaceList{}
+	if err := g.kube.Raw().List(ctx, nsl, &options); err != nil {
+		return "", fmt.Errorf("error setting resource: %w", err)
+	}
+
+	if len(nsl.Items) == 0 {
+		return "", ErrNamespaceNotFound
+	}
+
+	return nsl.Items[0].Name, nil
```

Comment:
We need to either error out of the list of namespaces > 1 or at least print a warning message with the NS you selected to use.  Assuming the first one is the right one will cause issues later.

Reply:
as mentioned before, there shouldn't be more than one, that's guaranteed in the installation, since this is not an external method for other parts of the code to consume, and belongs entirely to this business logic,  it's ok to assume the first one is the correct one.

Reply:
I'd still like to see a log message here that we are selecting the first NS in the list if there is >1 found.  :+1 to the go doc on the calling function but having a log message, in the unlikely event there is >1 NS with the label, will help us debug.  Optionally, if we are guaranteed to only have at most 1, then if we encounter more than one we are dealing with a badly configured cluster and will experience unknown issues so erroring out would be a good thing.

Reply:
fair enough, adding the error